core/vm/actor/builtin/v0/storage_power/storage_power_actor_export.cpp

```diff
@@ -46,13 +36,90 @@ namespace fc::vm::actor::builtin::v0::storage_power {
                 now,
                 event.miner_address,
                 common::hex_lower(event.callback_payload));
+
+            // Failures are unexpected here but will result in removal of miner
+            // power
+            auto maybe_claim = state.claims.tryGet(event.miner_address);
+            if (maybe_claim.has_error()) {
+              spdlog::warn(
+                  "failed to get claim for miner {} after failing "
+                  "OnDeferredCronEvent: {}",
+                  event.miner_address,
+                  maybe_claim.error().message());
+            } else {
+              if (!maybe_claim.value().has_value()) {
+                spdlog::warn(
+                    "miner OnDeferredCronEvent failed for miner %s with no "
+                    "power",
+                    event.miner_address);
+              } else {
+                if (state
+                        .addToClaim(event.miner_address,
+                                    -maybe_claim.value().get().raw_power,
+                                    -maybe_claim.value().get().qa_power)
+                        .has_error()) {
+                  spdlog::warn(
+                      "failed to remove ({}, {}) power for miner {} after to "
+                      "failed cron",
+                      maybe_claim.value().get().raw_power,
+                      maybe_claim.value().get().qa_power,
+                      event.miner_address);
+                }
+              }
+            }
           }
           return outcome::success();
         }));
         OUTCOME_TRY(state.cron_event_queue.remove(epoch));
       }
     }
-    state.last_epoch_tick = now;
+    state.first_cron_epoch = now + 1;
+
+    // charge gas to conform lotus implementation
+    OUTCOME_TRY(runtime.commitState(state));
+    OUTCOME_TRY(runtime.getCurrentActorStateCbor<State>());
+    OUTCOME_TRY(state.claims.hamt.loadRoot());
+    OUTCOME_TRY(runtime.commitState(state));
+
+    return outcome::success();
+  }
+
+  outcome::result<void> processBatchProofVerifiers(Runtime &runtime,
+                                                   State &state) {
+    if (state.proof_validation_batch.has_value()) {
+      OUTCOME_TRY(
+          verified,
+          runtime.verifyBatchSeals(state.proof_validation_batch.value()));
+      OUTCOME_TRY(miners, state.proof_validation_batch.value().keys());
+      for (const auto &miner : miners) {
+        auto found = verified.find(miner);
+        if (found == verified.end()) {
+          spdlog::warn("batch verify seals syscall implemented incorrectly");
+          return VMExitCode::kErrNotFound;
+        }
+        std::set<SectorNumber> successful;
```

Comment:
Must be `std::vector`

Reply:
I'm not sure, the order of method calls can change in miner actor, so I'll change it to std::vector.